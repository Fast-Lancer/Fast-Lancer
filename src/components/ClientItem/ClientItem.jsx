import styles from './ClientItem.module.css'
import phoneIcon from '../../assets/icons/phoneSmall.png'
import emailIcon from '../../assets/icons/emailSmall.png'

export default function ClientItem({ client }) {
  const {
    client_name: name,
    email,
    phone,
    business_name: businessName
  } = client

  return (
    <div className={styles.clientItem}>
      <div className={styles.col}>
        <h3 className={styles.h3}>
          {name}
        </h3>
        <h4 className={styles.h4}>
          {businessName}
        </h4>
      </div>
      <div className={styles.col2}>
        <div className={styles.row}>
          <img src={phoneIcon} alt='a phone icon' />
          {phone}
        </div>
        <div className={styles.row}>
          <img src={emailIcon} alt='a email icon' />
          {email}
        </div>
      </div>
    </div>

  )
}
