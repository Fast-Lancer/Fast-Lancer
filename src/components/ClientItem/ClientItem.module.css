.clientItem {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: #d3d3d3;
  padding: .2em;
  border-bottom: 1px solid black;
}

.col {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  max-width: 50%;
}

.row {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  gap: 5px;
}

.h3 {
  margin: .1em 0px;
  font-family: 'Playfair Display', serif;
  font-size: 28px;
}

.h4 {
  margin: .1em 0px;
  font-family: 'IBM Plex Mono', monospace;
  font-weight: 200;
  margin-left: .4em;
}

