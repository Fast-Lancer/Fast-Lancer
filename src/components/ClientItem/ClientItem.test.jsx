import { render } from '@testing-library/react'
import ClientItem from './ClientItem.jsx'

it('should render the ClientItem with the client info', () => {
  const { container } = render(
    <ClientItem client={{
      name: 'Bob Bobbertson',
      email: 'asdf@asdf.com',
      phone: '(360)-222-2222',
      business_name: 'Bobs Bobby Pins'
    }}/>
  )
  
  expect(container).toMatchSnapshot()
})
