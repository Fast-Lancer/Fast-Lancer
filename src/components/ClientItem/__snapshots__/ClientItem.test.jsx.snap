// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ClientItem with the client info 1`] = `
<div>
  <div
    class="clientItem"
  >
    <div
      class="col"
    >
      <h3
        class="h3"
      >
        Bob Bobbertson
      </h3>
      <h4
        class="h4"
      >
        Bobs Bobby Pins
      </h4>
    </div>
    <div
      class="col"
    >
      <div
        class="row"
      >
        <img
          alt="a phone icon"
          src="phoneSmall.png"
        />
        (360)-222-2222
      </div>
      <div
        class="row"
      >
        <img
          alt="a email icon"
          src="emailSmall.png"
        />
        asdf@asdf.com
      </div>
    </div>
  </div>
</div>
`;
